csharp/extractor/Semmle.Extraction/Context.cs

```diff
@@ -50,20 +50,30 @@ public SemanticModel Model(SyntaxNode node)
         /// <returns><code>true</code> iff the label already existed.</returns>
         public bool GetOrAddCachedLabel(ICachedEntity entity)
         {
-            var id = GetId(entity);
+            // Look up the label in the entityLabelCache
+            if (entityLabelCache.TryGetValue(entity, out Label label))
+            {
+                entity.Label = label;
+                return true;
+            }
+
+            var id = entity.Id;
+
             if (id == null)
                 throw new InternalError("Attempt to create a null entity for {0}", entity.GetType());
 
-            Label existingLabel;
-            if (labelCache.TryGetValue(id, out existingLabel))
+            // Look up the Id in the idLabelCache
+            if (idLabelCache.TryGetValue(id, out label))
```

Comment:
Why is `idLabelCache` needed? It appears to me that this is dead code, as `entityLabelCache.TryGetValue()` returns `false` iff `idLabelCache.TryGetValue()` returns `false`?

Reply:
After thinking about this, I see that this is indeed not dead code, but can be hit when two different entities share the same ID, and therefor label. Could you put a comment saying something like that here?